src/panel/PoiPanel.jsx

```diff
@@ -0,0 +1,237 @@
+/* globals _ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Telemetry from '../libs/telemetry';
+import nconf from '@qwant/nconf-getter';
+import layouts from './layouts.js';
+
+import ActionButtons from 'src/panel/poi/ActionButtons';
+import PoiHeader from 'src/panel/poi/PoiHeader';
+import PoiTitleImage from 'src/panel/poi/PoiTitleImage';
+import OpeningHour from 'src/components/OpeningHour';
+import OsmContribution from 'src/components/OsmContribution';
+import PoiBlockContainer from './poi_bloc/PoiBlockContainer';
+
+import CategoryService from '../adapters/category_service';
+import { openShareModal } from 'src/modals/ShareModal';
+
+import PoiStore from '../adapters/poi/poi_store';
+
+export default class PoiPanel extends React.Component {
+  static propTypes = {
+    poi: PropTypes.object.isRequired,
+    isFromFavorite: PropTypes.bool,
+    isFromCategory: PropTypes.bool,
+    sourceCategory: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.poi = props.poi;
+    if (props.poi) {
+      this.poi = PoiStore.deserialize(props.poi);
```

Comment:
Looks like it should be done outside of this component, so you can just assume here that `this.props.poi` is an object of the right kind.